### ISSUE TITLE
Add internal toggle for the new issue experience

### ISSUE THREAD

**Issue by malwilley** [getsentry/sentry#60282]
As a first step, we need to provide a simple toggle for users to opt in to the new experience.

Design specs:
- Button right of the ellipse menu in the top right hand part of the page
- Button icon: beaker
- Two states: 1) Off: no fill (default), 2) On: filled 

![Image](https://github.com/getsentry/sentry/assets/10888943/072b30fa-540c-4a0b-be15-61a9fe5d5acf)

![Image](https://github.com/getsentry/sentry/assets/10888943/872c670f-816d-4da4-b7cf-78a6cafaf5d9)

[Internal Figma](https://www.figma.com/file/799M4dxruJgBJd1OnfE0YA/Q4-Issue-Details-Layout?type=design&node-id=0-1&mode=design&t=mj3hl7vpIIEt0Iir-0)

Requirements:

- Should be available internally only (gated by Flagr)
- The toggle should use user settings, similar to the stack trace order and default event (recommended/latest)
